src/Param.cc

```diff
@@ -1209,21 +1576,29 @@ bool Param::ValidateValue() const
           {
             if (_val < std::get<T>(*this->dataPtr->minValue))
             {
-              sdferr << "The value [" << _val
-                     << "] is less than the minimum allowed value of ["
-                     << *this->GetMinValueAsString() << "] for key ["
-                     << this->GetKey() << "]\n";
+              std::ostringstream oss;
+              oss << _val;
+              std::string val_str = oss.str();
+              _errors.push_back({ErrorCode::PARAMETER_ERROR,
+                  "The value [" + val_str
+                  + "] is less than the minimum allowed value of ["
+                  + *this->GetMinValueAsString() + "] for key ["
+                  + this->GetKey() + "]"});
```

Comment:
nit: Since we're creating an `ostringstream`, we could just use it to construct the entire string and then call `err-rs.push_back({ErrorCode::PARAMETER_ERROR, oss.str()})`

Reply:
True, will do for both cases.